### PR TITLE
Docs: Improvements to doc builder

### PR DESCRIPTION
**Pull request by edg2s**
* Separate out project specific logic to make it more reusable in the future
* Make it 2.5x faster by batching lint-and-fix calls

### CODE REVIEW

File: tools/build-readme.js [hunk 1]
```diff
@@ -2,24 +2,29 @@ const fs = require( 'fs' );
 const template = fs.readFileSync( 'README.md.template', { encoding: 'UTF8' } );
 const allRules = require( '../src/index' ).rules;
 const rulesData = require( './rules-data' );
+const pluginName = 'no-jquery';
```
Comment (jdforrester) [inline tools/build-readme.js:5]: I'm sensing a factor-out coming. ;-)

Comment (edg2s) [reply tools/build-readme.js:5]: Now there's an idea...

Comment (jdforrester) [reply tools/build-readme.js:5]: eslint-plugin-meta? ;-)

Comment (edg2s) [reply tools/build-readme.js:5]: #229

**Pull request merged**